crates/bevy_app/src/app.rs

```diff
@@ -64,7 +64,17 @@ struct SubApp {
     app: App,
     runner: Box<dyn Fn(&mut World, &mut App)>,
 }
-
+impl Drop for App {
+    /// Gives a warning if the [`App`] is [`Drop`]ped without being run.
+    fn drop(&mut self) {
+        use bevy_utils::tracing::{span, warn, Level};
+        let _ = span!(Level::WARN, "bevy_app::App::drop").enter();
+        // Checks if there have been any ticks
+        if self.world.last_change_tick() == 0 {
```

Comment:
Maybe set a flag in `App::run` and check that either the last change tick is not 0 or this flag is set?

Reply:
We could also update the change ticks the first time app.update is run, which I think is my preference.

Reply:
That would require a more careful review to ensure there are no side effects